lecture_1/assignement/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sethgrid/pester"
+)
+
+type jobApiResponse struct {
+	Name   string
+	Age    int
+	Passed bool
+	Skills []string
+}
+
+const jobApi = "https://run.mocky.io/v3/f7ceece5-47ee-4955-b974-438982267dc8"
+
+func linearBackoff(retry int) time.Duration {
+	return time.Duration(retry) * time.Second
+}
+
+func fetchContent() []jobApiResponse {
+	httpClient := pester.New()
+	httpClient.Backoff = linearBackoff
```

Comment:
Using `pester` is not wrong per se but is kind-of an overkill for this. Using regular http client from the standard library is enough.

```suggestion
	httpClient := &http.Client{}
```

Reply:
Would it be appropriate to use http.Get() method to fetch the data since we do not need a stateful client for this task?

Reply:
Yep `http.Get()` works as well.